include/oneapi/dpl/pstl/hetero/dpcpp/sycl_defs.h

```diff
@@ -215,6 +216,28 @@ using __buffer_allocator =
     sycl::buffer_allocator;
 #endif
 
+template <typename _AtomicType, sycl::access::address_space _Space>
+struct __Atomic :
```

Comment:
Let's call it __atomic_ref, according to the semantics and our convensions.

Also, cannot it be made an alias instead, or there are significant API differences between SYCL classes that require unification on our level?

Reply:
Fixed: Let's call it __atomic_ref, according to the semantics and our conventions

Reply:
>>Also, cannot it be made an alias instead, or there are significant API differences between SYCL classes that require unification on our level?
- unfortunately yes, we have these differences: it is different values which we should pass into constructor of base class.

Reply:
OK, but then it should be semantically as close to sycl::atomic_ref as only possible. For example, maybe it can be defined as an alias for sycl::atomic_ref, and as a wrapper for the old sycl::atomic. Making it customized for only one particular use case may prevent its use in the future.

Reply:
For example, this is how I think it should be defined for SYCL 2020:

```
template <typename _AtomicType, sycl::memory_scope _Scope, sycl::access::address_space _Space>
#if _ONEDPL_SYCL2023_ATOMIC_REF_PRESENT
using __atomic_ref = sycl::atomic_ref<_AtomicType, sycl::memory_order::relaxed, _Scope, _Space>;
#else
struct __atomic_ref ...
#endif
```

Reply:
Fixed,